### PR TITLE
improved dataframe example to avoid confusion

### LINKED ISSUE #41180: DOC: DataFrame on df.nunique causing confusion
#### Documentation problem

The above documentation provides an example using `pd.DataFrame({'A': [1, 2, 3], 'B': [1, 1, 1]})` upon which `df.nunique()` returns 
```
A    3
B    1
``` 
This example might be a bit misleading as there are `3` and `1` present in dataframe as column elements and cause confusion. 
#### Suggested fix for documentation

I would like to  suggest using something like `pd.DataFrame({'A': [5, 6, 7], 'B': [5, 1, 1]})` upon which `df.nunique()` returns 
```
A 3
B 2
```
as this examples serves the purpose of `nunique` better as its clear that it counts the number of unique elements in column and does not return anything resembling any column element.

Furthermore, I would like to extend the example dataframe to `df.nunique(axis=1)` and match corresponding values with new dataframe as : 
```
df.nunique(axis=1)
0    1
1    2
2    2
dtype: int64
```

Does this change need to be documented in `whatsnew`?

### PR DESCRIPTION
**Pull request by regmibijay**
- [x] closes #41180
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -9827,10 +9827,10 @@ def nunique(self, axis: Axis = 0, dropna: bool = True) -> Series:
 
         Examples
         --------
-        >>> df = pd.DataFrame({'A': [1, 2, 3], 'B': [1, 1, 1]})
+        >>> df = pd.DataFrame({'A': [4, 5, 6], 'B': [4, 1, 1]})
```
Comment (dsaxton) [inline pandas/core/frame.py:9830]: Makes sense, could even use simple strings to make things blindingly obvious

Comment (regmibijay) [reply pandas/core/frame.py:9830]: I attempted to simplify the explaination string as well.

**Review by phofl — APPROVED** [review]
lgtm

**Review by attack68 — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @regmibijay. The CI failure looks unrelated

**Pull request merged**